src/wallet/spend.cpp

```diff
@@ -260,32 +260,24 @@ CoinsResult AvailableCoins(const CWallet& wallet,
             // Filter by spendable outputs only
             if (!spendable && only_spendable) continue;
 
+            // Obtain script type
+            std::vector<std::vector<uint8_t>> script_solutions;
+            TxoutType type = Solver(output.scriptPubKey, script_solutions);
+
             // If the Output is P2SH and spendable, we want to know if it is
             // a P2SH (legacy) or one of P2SH-P2WPKH, P2SH-P2WSH (P2SH-Segwit). We can determine
             // this from the redeemScript. If the Output is not spendable, it will be classified
             // as a P2SH (legacy), since we have no way of knowing otherwise without the redeemScript
-            CScript script;
             bool is_from_p2sh{false};
-            if (output.scriptPubKey.IsPayToScriptHash() && solvable) {
-                CTxDestination destination;
-                if (!ExtractDestination(output.scriptPubKey, destination))
-                    continue;
-                const CScriptID& hash = CScriptID(std::get<ScriptHash>(destination));
-                if (!provider->GetCScript(hash, script))
-                    continue;
+            if (type == TxoutType::SCRIPTHASH && solvable) {
```

Comment:
In the comment above it is mentioned that we check this if the output is "spendable". But in the conditional we are checking for "solvable", which I think is enough to serve the purpose. Should the comment be updated in that light?

Reply:
yeah. The comment isn't accurate.
Solvable refers to the wallet knowing how to create the unlocking script that spends the output (which is what we are looking for here), while spendable refers to whether the wallet has the key/s to sign the input or not.